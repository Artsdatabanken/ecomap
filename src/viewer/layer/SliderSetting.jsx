import React from 'react'
import { Slider, TextField } from 'material-ui'
import Setting from './Setting'
import getNext from '../../componentid'

const SliderSetting = ({title, icon, value, onChange}) =>
  <Setting>
    <div style={{position: 'relative', top: '21px', float: 'left'}}>{icon}</div>
    <div style={{position: 'relative', paddingLeft: '16px', top: '0px', height: '48px', width: '310px', float: 'left'}}>
      <TextField
        id={'tf' + getNext()}
        floatingLabelStyle={{position: 'absolute', top: '22px', left: '-40px'}}
        floatingLabelText={title}
        underlineShow={false}>
        <Slider
          min={0}
          max={1}
          step={0.05}
          value={value}
          onChange={(event, value) => onChange(value)}
          />
      </TextField>
    </div>
  </Setting>
export default SliderSetting
