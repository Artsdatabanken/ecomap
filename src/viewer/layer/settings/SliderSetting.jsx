import React from 'react'
import { Slider } from 'material-ui'
import Setting from './Setting'
import getNext from '../../../componentid'

const SliderSetting = ({ title, icon, value, onChange }) =>
  <Setting title={title} icon={icon}>
    <Slider
      sliderStyle={{marginTop: '6px', marginBottom: '0px'}}
      min={0}
      max={1}
      step={0.01}
      value={value}
      onChange={(event, value) => onChange(value)}
      id={getNext()}
      />
  </Setting>
export default SliderSetting
