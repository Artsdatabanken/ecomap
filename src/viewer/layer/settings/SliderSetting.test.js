import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import muiTheme from '../../theme_jest'

import SliderSetting from '../SliderSetting'

it('renders ok', () => {
  const tree = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <SliderSetting title='title' icon='' value={0.1} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
