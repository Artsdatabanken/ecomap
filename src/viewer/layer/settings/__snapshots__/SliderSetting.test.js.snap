// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "12px",
          "position": "relative",
        }
      }
    >
      title
    </div>
    <div
      style={
        Object {
          "float": "left",
          "position": "relative",
          "top": "2px",
        }
      }
    >
      
    </div>
    <div
      style={
        Object {
          "float": "left",
          "paddingBottom": "8px",
          "paddingLeft": "16px",
          "position": "relative",
          "width": "310px",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <div
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 18,
              "marginBottom": "0px",
              "marginTop": "6px",
              "muiPrepared": true,
              "position": "relative",
              "touchCallout": "none",
              "userSelect": "none",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "height": 2,
                "left": 0,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "[object Object]": 0,
                  "backgroundColor": "#00bcd4",
                  "directionInvariant": true,
                  "height": "100%",
                  "left": 0,
                  "marginRight": 6,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "calc(10%)",
                }
              }
            />
            <div
              style={
                Object {
                  "[object Object]": 0,
                  "backgroundColor": "#bdbdbd",
                  "directionInvariant": true,
                  "height": "100%",
                  "marginLeft": 6,
                  "muiPrepared": true,
                  "position": "absolute",
                  "right": 0,
                  "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "calc(90%)",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundClip": "padding-box",
                  "backgroundColor": "#00bcd4",
                  "border": "0px solid transparent",
                  "borderRadius": "50%",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "directionInvariant": true,
                  "height": 12,
                  "left": "10%",
                  "margin": "1px 0 0 0",
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "visible",
                  "pointerEvents": "inherit",
                  "position": "absolute",
                  "top": 0,
                  "transform": "translate(-50%, -50%)",
                  "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 12,
                  "zIndex": 1,
                }
              }
              tabIndex={0}
            >
              <div
                style={
                  Object {
                    "height": 12,
                    "left": 0,
                    "muiPrepared": true,
                    "overflow": "visible",
                    "position": "absolute",
                    "top": 0,
                    "width": 12,
                  }
                }
              />
            </div>
          </div>
        </div>
        <input
          max={1}
          min={0}
          name={undefined}
          required={true}
          step={0.05}
          type="hidden"
          value={0.1}
        />
      </div>
    </div>
  </div>
</div>
`;
